runtime/compiler/control/CompilationThread.cpp

```diff
@@ -9071,98 +9259,27 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
-               }
-
-            // There is the possibility that a hot/scorching compilation happened outside
-            // startup and with hints we move this expensive compilation during startup
-            // thus affecting startup time
-            // To minimize risk, add hot/scorching hints only if we are in startup mode
-            if (TR::Compiler->vm.isVMInStartupPhase(jitConfig))
-               {
-               TR_Hotness hotness = that->_methodBeingCompiled->_optimizationPlan->getOptLevel();
-               if (hotness == hot)
-                  {
-                  if (!isEDOCompilation)
-                     sc->addHint(method, TR_HintHot);
-                  }
-               else if (hotness == scorching)
-                  {
-                  sc->addHint(method, TR_HintScorching);
-                  }
-               // We also want to add a hint about methods compiled (not AOTed) during startup
-               // In subsequent runs we should give such method lower counts the idea being
-               // that if I take the time to compile method, why not do it sooner
-               sc->addHint(method, TR_HintMethodCompiledDuringStartup);
-               }
-            }
-
-         // If this is a cold/warm compilation that takes too much memory
-         // add a hint to avoid queuing a forced upgrade
-         // Look only during startup to avoid creating too many hints.
-         // If SCC is larger we could store hints for more methods
-         //
-         if (TR::Compiler->vm.isVMInStartupPhase(jitConfig))
-            {
-            if (scratchSegmentProvider.regionBytesAllocated() > TR::Options::_memExpensiveCompThreshold)
-               {
-               TR_Hotness hotness = that->_methodBeingCompiled->_optimizationPlan->getOptLevel();
-               if (hotness <= cold)
-                  {
-                  sc->addHint(method, TR_HintLargeMemoryMethodC);
-                  }
-               else if (hotness == warm)
-                  {
-                  sc->addHint(method, TR_HintLargeMemoryMethodW);
-                  }
-               }
-
-            if (that->getCompilationInfo()->getMethodQueueSize() - that->getQszWhenCompStarted() > TR::Options::_cpuExpensiveCompThreshold)
-               {
-               TR_Hotness hotness = that->_methodBeingCompiled->_optimizationPlan->getOptLevel();
-               if (hotness <= cold)
-                  {
-                  sc->addHint(method, TR_HintLargeCompCPUC);
-                  }
-               else if (hotness == warm)
-                  {
-                  sc->addHint(method, TR_HintLargeCompCPUW);
-                  }
-               }
-            }
-         }
+      TR_J9VMBase *fej9 = (TR_J9VMBase *)(compiler->fej9());
+      TR_CatchBlockProfileInfo * profileInfo = TR_CatchBlockProfileInfo::get(compiler);
+      storeHintsInTheSCC(compilationInfo, vm, fej9, metaData, profileInfo, scratchSegmentProvider);
```

Comment:
`compiler` may be NULL so we need to avoid executing the first two statements unconditionally.